frontend/views/Home.tsx

```diff
@@ -150,12 +286,117 @@ function Home(props: Props) {
             />
           );
         }}
-        //ListEmptyComponent={EmptyListMessage}
       />
     </SafeAreaView>
   );
 }
 
+/* ADMIN */
+
+/**
+ * Properties for {@link Users}
+ */
+type AdminProps = {
+  // callback called when the {@link FlatList} is refreshed
+  onRefresh: () => void;
+  // Users to render
+  users: { name: string; email: string; superUser: boolean }[];
+  navigation: any;
+  token: string;
+  fetchData: Function;
+};
+
+/**
+ * Component that displays a list of components of either {@link Users}
+ *
+ * @param {AdminProps} props
+ * @returns {React.ReactElement} React component
+ */
+function Admins(props: AdminProps): React.ReactElement {
+  const { users, onRefresh } = props;
+  const [refreshing, setRefreshing] = useState(false);
+  const [visible, setVisible] = React.useState(false);
+
+  const onDelete = (email: string) => {
+    //TODO: not refreshing the data on client side
+    Admin.delete(email, props.token);
+    props.fetchData();
+  };
+
+  const onEdit = (user: {
+    email: string;
+    name: string;
+    superUser: boolean;
+  }) => {
+    //TODO: update Admin.update() perams
+    //Admin.update(email, props.token)
+    props.fetchData();
+  };
+
+  const onInvite = (email: string, token: string) => {
+    Admin.invite(email, token);
```

Comment:
this should be awaited and error handled. Ie:

```ts
    try {
        await Admin.invite(email, token);
    } catch (err) {
        // somehow display error
    }
        
```

This requires that the `onInvite` function has to be async. Also unnecessary to pass `token` since it's in the current scope and is available to this funciton.

```ts
const onInvite = async (email: string) => { /* ... */ };
```